src/Tooltip/Tooltip.component.tsx

```diff
@@ -0,0 +1,255 @@
+    font-family: ${th('typography.fontFamily')};
+    font-size: 0.8rem;
+    padding: 0.5rem;
+    opacity: 0;
+    transition: opacity 250ms ease-in-out;
+
+    ${({ wrap = true }) =>
+        wrap ? css({ whiteSpace: 'normal' }) : css({ whiteSpace: 'nowrap' })};
+
+    ${({ maxWidth = 'auto' }) => css({ width: maxWidth })};
+
+    ${({ position, height, width, toolTipHeight, toolTipWidth }) =>
+        positionVariants(position, height, width, toolTipHeight, toolTipWidth)};
+`;
+
+export class Tooltip extends React.Component<
+    TooltipContainerProps,
+    TooltipContainerState
+> {
+    public state = {
+        width: 0,
+        height: 0,
+        toolTipHeight: 0,
+        toolTipWidth: 0,
+        hidden: true,
+    };
+    private readonly tooltipContainerRef: React.RefObject<
+        HTMLDivElement
+    > = React.createRef();
+    private readonly tooltipRef: React.RefObject<
+        HTMLDivElement
+    > = React.createRef();
+
+    componentDidMount(): void {
+        const { current: toolTipContainer } = this.tooltipContainerRef;
+        const { current: toolTip } = this.tooltipRef;
+        // TODO: instead of setting state, what about using the render fxn?
+        if (toolTipContainer) {
+            this.setState({
+                height: toolTipContainer.clientHeight,
+                width: toolTipContainer.clientWidth,
+            });
+        }
+        if (toolTip) {
+            this.setState({
+                toolTipHeight: toolTip.clientHeight,
+                toolTipWidth: toolTip.clientWidth,
+            });
+        }
+    }
+
+    render(): React.ReactElement<Tooltip> {
+        const {
+            className,
+            children,
+            content,
+            position,
+            wrap,
+            background,
+            color,
+            maxWidth,
+            ...props
+        } = this.props;
+        const { height, width, toolTipHeight, toolTipWidth } = this.state;
+        return (
+            <ToolTipContainer
+                content={content}
+                onMouseEnter={() => this.setState({ hidden: false })}
+                onMouseLeave={() => this.setState({ hidden: true })}
+                ref={this.tooltipContainerRef}
+                className={classNames('anchor-tooltip', className)}
+                {...props}
+            >
+                {children}
+                <TooltipElement
+                    className="anchor-tooltip-element"
+                    ref={this.tooltipRef}
+                    position={position || 'topEnd'}
+                    height={height}
+                    width={width}
+                    toolTipHeight={toolTipHeight}
+                    toolTipWidth={toolTipWidth}
+                    children={content}
+                    wrap={wrap}
+                    background={background}
+                    color={color}
+                    maxWidth={maxWidth}
+                    style={{ opacity: this.state.hidden ? 0 : 1 }}
```

Comment:
Since we're avoiding inline styles, you could instead add the following to `TooltipElement`...
```
    &.active {
        opacity: 1;
        visibility:visible;
    }
```

And change line 238 to:

`className={classNames('anchor-tooltip-element', !this.state.hidden && 'active')}`

Reply:
I want to preserve the animation, so this won't solve that.